covalent/executor/base.py

```diff
@@ -493,6 +510,14 @@ async def execute(
 
         return (result, stdout.getvalue(), stderr.getvalue())
 
+    async def setup(self):
```

Comment:
Are you planning to integrate `setup` and `teardown` with `execute()`?

Reply:
the dispatcher should call setup and teardown before/after execute, if they exist